### PR TITLE
Creating chromatic baseline for fixing combobox to use useFormProps

### LINKED ISSUE #2323: ComboBox Form context inheritance issue with label props
# 🐛 Bug Report 

Side label end doesn't position correctly. Root cause is form context isn't being set in ComboBox, i.e useFormProps.

## 🤔 Expected Behavior

ComoBox inherits all the props from a From.

## 😯 Current Behavior

The context isn't being passed correctly to ComboBox so label aren't being positioned correctly.

## 💁 Possible Solution

Adding Form context, useFormProps, to ComboBox or refactoring Form context to be in Field.tsx so the individual components don't have each inherit. https://github.com/adobe/react-spectrum/search?q=useFormProps

## 🔦 Context

This was noticed in a Form with only comboboxes.

## 💻 Code Sample

https://codesandbox.io/s/heuristic-pascal-i1hmy
https://codesandbox.io/s/vibrant-driscoll-c6697?file=/src/App.tsx:199-1287

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.12.x and 3.13.x |

## 🧢 Your Company/Team
RSP

### PR DESCRIPTION
**Pull request by ktabors**
Closes #2323

Chromatic updates for #2351 
Adding more from components to form and provider storybook and chromatic.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Review the form stories
Review the provider stories

Run chromatic

## 🧢 Your Project:
RSP

### CODE REVIEW

File: packages/@react-spectrum/form/stories/Form.stories.tsx [hunk 1]
```diff
@@ -150,6 +153,13 @@ function render(props: any = {}) {
         <Checkbox value="cats">Cats</Checkbox>
         <Checkbox value="dragons">Dragons</Checkbox>
       </CheckboxGroup>
+      <ColorField label="Primary Color" />
```
Comment (ktabors) [inline packages/@react-spectrum/form/stories/Form.stories.tsx:156]: May I refactor this to be one component of each type and in alphabetical order? Ditto the Provider update below.

File: packages/@react-spectrum/form/chromatic/Form.chromatic.tsx [hunk 2]
```diff
@@ -50,6 +54,20 @@ const Template = (): Story<SpectrumFormProps> => (args) => (
       <Radio value="dragons">Dragons</Radio>
     </RadioGroup>
     <TextArea label="Comments" placeholder="How do you feel?" />
+    <ColorField label="Primary Color" />
```
Comment (ktabors) [inline packages/@react-spectrum/form/chromatic/Form.chromatic.tsx:57]: Added at the bottom to try to make the diff change less pronounced. Ditto the Provider update below.

File: packages/@react-spectrum/provider/chromatic/Provider.chromatic.tsx [hunk 3]
```diff
@@ -38,6 +45,48 @@ storiesOf('Provider', module)
   .add(
     'custom theme',
     () => render({theme: THEME})
+  )
```
Comment (ktabors) [inline packages/@react-spectrum/provider/chromatic/Provider.chromatic.tsx:48]: Should this file be refactored to use the new syntax?

**Review by ktabors — COMMENTED** [review]
Need to review chromatic before this PR is ready.

File: packages/@react-spectrum/form/chromatic/Form.chromatic.tsx (hunk 2 above)
Comment (snowystinger) [inline packages/@react-spectrum/form/chromatic/Form.chromatic.tsx:57]: lets not add colorfield to forms for chromatic, it's not correct currently i believe

File: packages/@react-spectrum/provider/stories/Provider.stories.tsx [hunk 4]
```diff
@@ -173,13 +180,37 @@ function render(props = {}) {
...
+          <Picker name="favorite-color3" label="Favorite color searchwithin">
+            <Item key="red">Red</Item>
+            <Item key="orange">Orange</Item>
+            <Item key="yellow">Yellow</Item>
+            <Item key="green">Green</Item>
+            <Item key="blue">Blue</Item>
+            <Item key="purple">Purple</Item>
+          </Picker>
+        </SearchWithin>
+        <Slider label="Cookies to buy" defaultValue={12} />
```
Comment (snowystinger) [inline packages/@react-spectrum/provider/stories/Provider.stories.tsx:213]: slider is pretty bad in forms currently as well

File: packages/@react-spectrum/textfield/src/TextFieldBase.tsx [hunk 5]
```diff
@@ -39,8 +37,6 @@ interface TextFieldBaseProps extends Omit<SpectrumTextFieldProps, 'onChange'>, P
 }
 
 function TextFieldBase(props: TextFieldBaseProps, ref: Ref<TextFieldRef>) {
-  props = useProviderProps(props);
```
Comment (snowystinger) [inline packages/@react-spectrum/textfield/src/TextFieldBase.tsx:42]: why was useProviderProps removed here but not in SearchWithin? or should it be the other way around?

File: packages/@react-spectrum/form/stories/Form.stories.tsx (hunk 1 above)
Comment (snowystinger) [reply packages/@react-spectrum/form/stories/Form.stories.tsx:156]: I mean, it's already mangling the chromatic story, so, i guess? Do be careful, form chromatic stories were already close to their maximum height
It falls into the same category as the other chromatic PR you have, we just need to do it in three steps very carefully
first run chromatic on main, make sure it's all green. then run it on this branch, make sure we're ok with everything. merge it, then run chromatic once more on main so we have the right baseline moving forward

File: packages/@react-spectrum/provider/chromatic/Provider.chromatic.tsx (hunk 3 above)
Comment (snowystinger) [reply packages/@react-spectrum/provider/chromatic/Provider.chromatic.tsx:48]: at some point yes, if it's not hard you can do it here, if it takes more than a few min, maybe do it on its own later

File: packages/@react-spectrum/textfield/src/TextFieldBase.tsx (hunk 5 above)
Comment (ktabors) [reply packages/@react-spectrum/textfield/src/TextFieldBase.tsx:42]: In my exploration I found having useProviderProps where did nothing. And it was needed in SearchWithin for the props like isQuiet, isRequired, isDisabled, etc. It is needed in any components that inherit those from the Provider. I was annoyed that I couldn't move useProviderProps to Field too.

File: packages/@react-spectrum/form/chromatic/Form.chromatic.tsx (hunk 2 above)
Comment (ktabors) [reply packages/@react-spectrum/form/chromatic/Form.chromatic.tsx:57]: It looked pretty good to me, but I'll remove it.

File: packages/@react-spectrum/form/stories/Form.stories.tsx (hunk 1 above)
Comment (ktabors) [reply packages/@react-spectrum/form/stories/Form.stories.tsx]: This sounds good. I was hoping that adding them at the end wouldn't cause too much chaos, but after reviewing, adding this many components makes the diff basically useless. :(

How do I tell what the max height is? Should the form stories be separated by categories like inputs and dropdowns?

**Review by snowystinger — COMMENTED** [review]
So it's occurred to me, we're doing code changes and chromatic changes together in this PR. They need to be split since they are modifications that will cause us to not get anything useful out of chromatic if kept together. Lets get the chromatic modifications out into their own PR, we'll merge that one, then we can merge the refactor of useFormProps

same plan as we did for the locale updates you were doing

**Comment by ktabors** [comment]
> So it's occurred to me, we're doing code changes and chromatic changes together in this PR. They need to be split since they are modifications that will cause us to not get anything useful out of chromatic if kept together. Lets get the chromatic modifications out into their own PR, we'll merge that one, then we can merge the refactor of useFormProps
> 
> same plan as we did for the locale updates you were doing

#2351 has the code change, updating this to be chromatic and story updates

**Review by LFDanLu — APPROVED** [review]
LGTM, should take special care when approving the chromatic diffs/new baseline for this however since this changes the order of the components in these stories

**Comment by dannify** [comment]
Can we keep the order of the form elements as they were before to minimize the visual diff for this?

**Comment by ktabors** [comment]
> Can we keep the order of the form elements as they were before to minimize the visual diff for this?

That was my plan, but I found that by adding components it makes keeping the order pointless. The form width is different with these other components and after the first iteration, all the variations clobber each other because the height of each form is longer. I found more value in them being alphabetical to make sure we have all the ones we want.

**Comment by dannify** [comment]
ok sure.

**Review by reidbarber — APPROVED** [review]
LGTM

**Pull request merged**